cadc-download-manager/src/main/java/ca/nrc/cadc/dlm/DownloadTupleFormat.java

```diff
@@ -101,83 +107,88 @@ public DownloadTuple parse(String tupleStr) throws DownloadTupleParsingException
-                // trim off trailing "}".
-                // guaranteed to be there due to
-                // check for equal occurrences of { and } above.
-                tmpLabel = convertLabelText(l.substring(0, l.length() - 1));
-            } else {
-                // invalid format
-                throw new DownloadTupleParsingException("invalid label format: " + tupleStr);
+        String[] tupleParts = tupleStr.split("\\{");
+        // Get tuple URI - should at least have this.
+        URI tmpURI = null;
+        String uriStr = tupleParts[0];
+        log.debug("uri given: " + uriStr);
+        if (StringUtil.hasLength(uriStr)) {
+            try {
+                tmpURI = new URI(uriStr);
+            } catch (URISyntaxException u) {
+                throw new DownloadTupleParsingException("id parsing error: " + u + ": " + tupleStr);
             }
         } else {
-            tmpLabel = null;
+            // invalid format - has to at least be a single URI passed in
+            throw new DownloadTupleParsingException("zero length id found: " + tupleStr);
         }
 
-        // Get any cutout that might be there
-        Shape tmpShape = null;
-        if (tupleParts.length > 1) {
-            String sd = tupleParts[1];
-            if (sd.length() > 1) {
+        // Create new DownloadTuple
+        DownloadTuple newTuple = new DownloadTuple(tmpURI);
+
+        if (openBraceCount == 3) {
+            String tmpPosStr = tupleParts[1];
+            if (tmpPosStr.length() > 1) {
                 // trim off trailing "}"
-                // guaranteed to be there due to
-                // check for equal occurrences of { and } above.
-                String tmpShapeStr = sd.substring(0, sd.length() - 1);
-                // put off parsing until id is parsed
+                String tmpShapeStr = tmpPosStr.substring(0, tmpPosStr.length() - 1);
                 log.debug("cutout string: " + tmpShapeStr);
                 if (StringUtil.hasLength(tmpShapeStr)) {
                     try {
                         ShapeFormat sf = new ShapeFormat();
-                        tmpShape = sf.parse(tmpShapeStr);
+                        newTuple.posCutout = sf.parse(tmpShapeStr);
                         log.debug("parsed cutout.");
                     } catch (IllegalArgumentException ill) {
                         log.debug("parsing error for cutout: " + tmpShapeStr);
-                        throw new DownloadTupleParsingException("cutout parsing error: " + ill + ": " + tupleStr);
+                        throw new DownloadTupleParsingException("pos cutout parsing error: " + ill + ": " + tmpPosStr);
                     } catch (Exception e) {
-                        log.debug("other error for parsing cutout:" + e);
-                        throw new DownloadTupleParsingException("BUG for cutout:" + e + ": " + tupleStr);
+                        log.debug("other error for parsing pos cutout:" + e);
```

Comment:
Use the Exception argument to `log.debug` (i.e. `log.debug("...", e)`) rather than assembling it into the message to let Log4J handle how it should be set.

Reply:
Changed both of these where it applied through the file.